### PR TITLE
Add sentry integration

### PR DESCRIPTION
**Pull request by RJ722**
Closes: #14

### CODE REVIEW

File: init.go [hunk 1]
```diff
@@ -2,17 +2,28 @@ package main
...
+var SentryDSN string
+
+func run() int {
 	// Print default stub message
 	fmt.Printf("DeepSource Command Line Interface " + cliVersion + "\n \n")
 
+	sentry.Init(sentry.ClientOptions{
+		Dsn: SentryDSN,
+	})
+	fmt.Println(SentryDSN)
```
Comment (jaipradeesh) [inline init.go:25]: Sentry's DSN needn't be printed.

File: init.go [hunk 2]
```diff
@@ -2,17 +2,27 @@ package main
...
-func main() {
+var SentryDSN string
+
+func run() int {
 	// Print default stub message
 	fmt.Printf("DeepSource Command Line Interface " + cliVersion + "\n \n")
 
+	sentry.Init(sentry.ClientOptions{
+		Dsn: SentryDSN,
+	})
```
Comment (sourya) [inline init.go:24]:
```suggestion
	err := sentry.Init(sentry.ClientOptions{
		Dsn: SentryDSN,
	})
	
	if err != nil {
		fmt.Println("Could not load sentry.")
		return 1
	}
```

File: init.go [hunk 3]
```diff
@@ -2,17 +2,32 @@ package main
...
 	// Print default stub message
 	fmt.Printf("DeepSource Command Line Interface " + cliVersion + "\n \n")
 
+	err := sentry.Init(sentry.ClientOptions{
+		Dsn: SentryDSN,
+	})
+	
+	if err != nil {
+		fmt.Println("Could not load sentry.")
+		return 1
```
Comment (RJ722) [inline init.go:28]:
```suggestion
```

File: init.go [hunk 4]
```diff
@@ -46,33 +56,36 @@ func main() {
...
-		os.Exit(1)
+		return 1
 	}
 
 	// Get current path
 	currentDir, err := os.Getwd()
 	if err != nil {
 		fmt.Println("DeepSource | Error | Unable to identify current directory.")
-		os.Exit(0)
+		sentry.CaptureException(err)
```
Comment (sourya) [inline init.go:70]: How would we know more about the user? Log the DEEPSOURCE_DSN and the args in sentry, too.

File: init.go [hunk 5]
```diff
@@ -84,8 +97,10 @@ func main() {
 
 	// Check for valid protocol
 	if strings.HasPrefix(dsnSplitProtocolBody[0], "http") == false {
-		fmt.Println("DeepSource | Error | DSN specified should start with http(s). Cross verify DEEPSOURCE_DSN value against the settings page of the repository.")
-		os.Exit(1)
+		errMessage := "DeepSource | Error | DSN specified should start with http(s). Cross verify DEEPSOURCE_DSN value against the settings page of the repository."
+		fmt.Println(errMessage)
+		sentry.CaptureException(errors.New(errMessage))
```
Comment (sourya) [inline init.go:106]:
```suggestion
		err := errors.New("DeepSource | Error | DSN specified should start with http(s). Cross verify DEEPSOURCE_DSN value against the settings page of the repository.")
		fmt.Println(err)
		sentry.CaptureException(err)
```

File: init.go [hunk 6]
```diff
@@ -212,18 +233,27 @@ func main() {
...
-		os.Exit(0)
+		return 0
 	}
+	return 0
+}
+
+func main() {
+	returnCode := run()
+	defer func() { os.Exit(returnCode) }()
+	defer sentry.Flush(2 * time.Second)
```
Comment (sourya) [inline init.go:262]: 👍

File: init.go (hunk 4 above)
Comment (RJ722) [reply init.go:82]: Nice catch. Done!

**Pull request merged**